src/opera/pge/base_pge.py

```diff
@@ -238,24 +241,62 @@ def _run_sas_qa_executable(self):
             self.logger.info(self.name, ErrorCode.QA_SAS_PROGRAM_DISABLED,
                              'SAS QA is disabled, skipping')
 
+    def _checksum_output_products(self):
+        """
+        Generates a dictionary mapping output product file names to the
+        corresponding MD5 checksum digest of the file's contents.
+
+        The output products to generated checksums for is determined by scanning
+        the output product location specified by the RunConfig. Any files
+        within the directory that have the expected file extensions for output
+        products are then picked up for checksum generation.
+
+        Returns
+        -------
+        checksums : dict
+            Mapping of output product file names to MD5 checksums of said
+            products.
+
+        """
+        output_products = self.runconfig.get_output_product_filenames()
+
+        checksums = {
+            basename(output_product): get_checksum(output_product)
+            for output_product in output_products
+            if get_extension(output_product) in self.rename_by_extension_map
```

Comment:
These 2 lines (265 and 266) are not clear to me.  I realize you are making the checksums dictionary but I have not see for statements and if statements that do not end in colon.  It's very concise but not very clear to me